### PR TITLE
ui: Node list view redesign - Fix follow up issues

### PR DESCRIPTION
**Pull request by koorosh**
Current PR contains fixes on following issues: https://github.com/cockroachdb/cockroach/issues/43883

- Dropdown component refactored to include default button as a toggle component and
added render prop to provide own implementation of toggle component
- Button components are extended with ability to render Icon within the button
- Fixed stylings on Cluster overview page

### CODE REVIEW

**Comment by dhartunian** [comment]
@Koorosh sorry I commented on issue and not the PR. Can you rebase this and make the small change to the triangle icon on menu as clarified by Anne in the issue?

**Comment by Koorosh** [comment]
> @Koorosh sorry I commented on issue and not the PR. Can you rebase this and make the small change to the triangle icon on menu as clarified by Anne in the issue?

Done.

**Comment by Annebirzin** [comment]
@Koorosh This is looking great! I noticed a few tiny adjustments that could be made (these are not critical and can be a separate issue if needed)

**Node statuses:**
- [ ] Make all node status fonts bold

**Node names:**
- [ ] Make all node numbers bold - just the number not the entire name (ie. **N1** us-east1)

**Node list module:**
- [ ] Add space below the last node and the white bound box (so there is some white space below the list) 
Screen Shot 2020-02-05 at 4 04 07 PM

**Node list/map selection bar**
- [ ] The height of the bar seems to change when I switch from list to map. Wondering if we can keep these at the same height (using the node list height)

**Node/decommissioned node table headers:**
- [ ] the font still seems to be ‘regular’ not ‘semi-bold’ (ideally we can match the statement page table header)
- [ ] node table header: the background grey should be #F5F7FA 
- [ ] update ‘# of nodes’ column header to ‘node count’ 
Screen Shot 2020-02-05 at 3 21 52 PM

**Comment by koorosh** [comment]
@Annebirzin , @danhhz 
Done, adjustments are included in this PR.

**Comment by Annebirzin** [comment]
@koorosh I think you meant to tag @dhartunian ?

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 1]
```diff
@@ -413,7 +424,7 @@ const decommissionedNodesTableData = createSelector(
   partitionedStatuses,
   nodesSummarySelector,
   (statuses, nodesSummary): DecommissionedNodeStatusRow[] => {
-    const decommissionedStatuses = statuses.decommissioned || [];
+    const decommissionedStatuses = statuses.live || [];
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: I think this modification is incorrect and it should remain `decommissioned`. Can you take another look?

Maybe add a test to check :)

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 2]
```diff
@@ -133,8 +133,9 @@ class NodeList extends React.Component<LiveNodeListProps> {
         if (!!record.nodeId) {
           return (
             <React.Fragment>
-              <Link to={`/node/${record.nodeId}`}>
-                <Text>{record.nodeId}</Text>
+              <Link className="nodes-table__link" to={`/node/${record.nodeId}`}>
+                <Text>{`N`}</Text>
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Can you take another look at Anne's example? I think the `N` and the number are bolded together.

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 3]
```diff
@@ -153,6 +154,7 @@ class NodeList extends React.Component<LiveNodeListProps> {
         return 0;
       },
       className: "column--border-right",
+      width: "20%",
     },
     {
       key: "nodesCount",
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Last item on Anne's list was adjustment of this column's naming. I think that's the line below this.

**Review by dhartunian — CHANGES_REQUESTED** [review]
Thanks for the updates @koorosh. A few small adjustments remaining. Let me know if you have any questions.

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx (hunk 2 above)
Comment (koorosh) [reply pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Thank you, fixed.

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx (hunk 3 above)
Comment (koorosh) [reply pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: You're right. Renamed column name.

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx (hunk 1 above)
Comment (koorosh) [reply pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: 🤦‍♂️oh, that's a terrible result of manual testing where I used "live" nodes to emulate some data in Decommissioned nodes list, and haven't reverted code back.
It is a good point to add tests on these selectors.

**Review by dhartunian — APPROVED** [review]

**Pull request merged**